test/e2e/user_quota_tracing/user_qauta_tracing_test.go

```diff
@@ -0,0 +1,174 @@
+			GroupsNames            []string
+		}{
+			{"teacher-app-01", queuePath[1], users[0], []string{groups[0]}},
+			{"students-app-01", queuePath[2], users[1], []string{groups[1]}},
+			{"assistant-app-01", queuePath[2], users[2], []string{groups[1]}},
+			{"assistant-app-02", queuePath[1], users[2], []string{groups[0]}},
+		}
+		for _, config := range configs {
+			By(fmt.Sprintf("Deploy the sleep app %s to the %s namespace", config.AppID, ns))
+			sleepObj, podErr := k8s.InitSleepPod(k8s.SleepPodConfig{
+				AppID:  config.AppID,
+				NS:     ns,
+				Labels: map[string]string{"queue": config.Queue},
+			})
+			sleepObj.ObjectMeta.Annotations["yunikorn.apache.org/user.info"] = fmt.Sprintf("{username:%s, groups:{%s}}", config.UserName, config.GroupsNames)
+			Ω(podErr).NotTo(HaveOccurred())
+			sleepRespPod, podErr := kClient.CreatePod(sleepObj, ns)
+			Ω(podErr).NotTo(HaveOccurred())
+
+			// Wait for pod to move to running state
+			podErr = kClient.WaitForPodBySelectorRunning(ns,
+				fmt.Sprintf("app=%s", sleepRespPod.ObjectMeta.Labels["app"]), WAIT_INTERVAL)
+			Ω(podErr).NotTo(HaveOccurred())
+		}
+		restClient := yunikorn.RClient{}
+		var usedResource yunikorn.ResourceUsage
+		By("Check total number of users and groups")
+		usersUsage, err := restClient.GetUsersResourceUsage(DEFAULT_PARTITION)
+		Ω(err).NotTo(HaveOccurred())
+		Ω(len(usersUsage)).To(Equal(3), "Total number of users is not correct")
+		groupsUsage, err := restClient.GetGroupsResourceUsage(DEFAULT_PARTITION)
+		Ω(err).NotTo(HaveOccurred())
+		Ω(len(groupsUsage)).To(Equal(2), "Total number of groups is not correct")
+
+		By("Check user resource usage of assistant in each queue")
+		userUsage, err := restClient.GetUserResourceUsage(DEFAULT_PARTITION, users[2])
+		Ω(err).NotTo(HaveOccurred())
+		Ω(userUsage).To(Equal(map[string]string{configs[2].AppID: groups[1], configs[3].AppID: groups[0]}))
+		queue, err := restClient.GetQueueFromUserResourceUsage(usersUsage, queuePath[0], users[2])
+		Ω(err).NotTo(HaveOccurred())
+		Ω(queue.RunningApplications).To(Equal([]string{configs[2].AppID, configs[3].AppID}))
+		usedResource.ParseResourceUsage(parseResource(queue.ResourceUsage))
+		Ω(usedResource).To(Equal(map[string]int64{siCommon.CPU: 200, siCommon.Memory: 100}))
+		queue, err = restClient.GetQueueFromUserResourceUsage(usersUsage, queuePath[1], users[2])
+		Ω(err).NotTo(HaveOccurred())
+		Ω(queue.RunningApplications).To(Equal([]string{configs[3].AppID}))
+		usedResource.ParseResourceUsage(parseResource(queue.ResourceUsage))
+		Ω(usedResource).To(Equal(map[string]int64{siCommon.CPU: 100, siCommon.Memory: 50}))
+		queue, err = restClient.GetQueueFromUserResourceUsage(usersUsage, queuePath[2], users[2])
+		Ω(err).NotTo(HaveOccurred())
+		Ω(queue.RunningApplications).To(Equal([]string{configs[2].AppID}))
+		usedResource.ParseResourceUsage(parseResource(queue.ResourceUsage))
+		Ω(usedResource).To(Equal(map[string]int64{siCommon.CPU: 100, siCommon.Memory: 50}))
```

Comment:
Can we organize asserts one by one? first all **user** tracker rest api response assertions and then **users** tracker rest api response assertions.

Similarly for group and groups